src/dmd/traits.d

```diff
@@ -1842,6 +1842,25 @@ Lnext:
         auto tup = new TupleExp(e.loc, exps);
         return tup.expressionSemantic(sc);
     }
+    if (e.ident == Id.getUniqueIdentifier)
+    {
+        if (dim != 1)
+            return dimError(1);
+        auto ex = isExpression((*e.args)[0]);
+        if (ex)
+        {
+            if (StringExp se = ex.ctfeInterpret().toStringExp())
+            {
+                StringExp s = new StringExp(e.loc, cast(char*) (Identifier.generateId(se.toStringz.ptr).toChars()));
+                return s.expressionSemantic(sc);
+            }
+        }
+        if (ex)
+            e.error("`getUniqueIdentifier` parameter must give a string, not `%s`", ex.toChars());
```

Comment:
```suggestion
            e.error("`getUniqueIdentifier` parameter must be a string, not `%s`", ex.toChars());
```

Reply:
I have employed "give" on purpose because it can be a CTFE function too. Also there's a second error message and the test case in fail_compilation needs update.

Reply:
How about "evaluate to"